tree/tree/src/TTree.cxx

```diff
@@ -1209,6 +1209,18 @@ void TTree::AddClone(TTree* clone)
    }
 }
 
+void CheckIfReshuffled(TTree *t1, TTree *t2)
+{
+   const auto isT1Reshuffled = t1->TestBit(TTree::kEntriesReshuffled);
+   const auto isT2Reshuffled = t2->TestBit(TTree::kEntriesReshuffled);
+   if (isT1Reshuffled || isT2Reshuffled) {
```

Comment:
```suggestion
   if ((isT1Reshuffled || isT2Reshuffled) && !t2->GetTreeIndex())
```

Reply:
Thanks @pcanal , one question: any of the tree that is being added as a friend or the tree that is adding a friend could be the reshuffled tree. why are we only ok if `t2` (the tree that is adding the friend, with the current implementation) has a tree index? It could still be that the entries reshuffled are the ones of the other tree...?

Reply:
>  It could still be that the entries reshuffled are the ones of the other tree...?

Yes but the main tree *always* determine the order of reading.

Reply:
Now also testing for TTreeIndex